pkg/bee/api/api.go

```diff
@@ -237,21 +239,6 @@ func (c *Client) requestWithHeader(ctx context.Context, method, path string, hea
 	return err
 }
 
-// drain discards all of the remaining data from the reader and closes it,
-// asynchronously.
-func drain(r io.ReadCloser) {
-	go func() {
-		// Panicking here does not put data in
-		// an inconsistent state.
-		defer func() {
-			_ = recover()
-		}()
-
-		_, _ = io.Copy(ioutil.Discard, r)
-		r.Close()
```

Comment:
This is where http body is closed. The drain function additionally reads all remaining body as the connection is not closed as the body is not read completely.

Reply:
`Close()` is normally enough, no draining is needed. The issue the PR tries to solve is that there were missing Close statements.

Reply:
Normally the `Close()` is enough, no need to drain. The issue at hand is that the Close is not called in a few places.

Reply:
Actually, there are reports that draining is needed in order to reuse connection. Some sources: [here](https://forum.golangbridge.org/t/do-i-need-to-read-the-body-before-close-it/5594/11), [here](https://github.com/golang/go/issues/15703) and [here](https://go-review.googlesource.com/c/go/+/21290/). There is a possibility to create TIME_WAIT connections if the response body is not read. This prevents that possibility.